src/crypto.js

```diff
@@ -40,12 +40,10 @@ exports.getRandomValues = function (buffer) {
     view = new Uint8Array(buffer),
     i;
 
-  if (offset + size > buf.length) {
-    throw new Error("Insufficient Randomness Allocated.");
-  }
-
+  // Hack: just wrap around.
```

Comment:
I think that would unnecessarily add hidden to this system (which is there now, but should get removed).

When would you know when the trial ends?

I found this while writing https://github.com/uProxy/rappor-test-client/blob/master/test-client.js, which creates an array of metrics and aggregates them into a single JSON (simulating multiple senders of the metrics).  I don't know when you'd be able to tell when the client is done queueing all the promises for metric.report(..) before you'd get enough information on how much random entropy you need.